### DISCUSSION TITLE
Whoops，看来我们遇到了一个问题：invalid_client

### DISCUSSION THREAD

**Discussion by tanix12qly** [spencerwooo/onedrive-vercel-index#1052] [Q&A] [answered]
help!!!

AADSTS7000222: The provided client secret keys for app 'd87bcc39-1750-4ca0-ad54-f8d0efbb2735' are expired. Visit the Azure portal to create new keys for your app: https://aka.ms/NewClientSecret, or consider using certificate credentials for added security: https://aka.ms/certCreds.

Trace ID: 541142d3-332e-49d0-9ab4-a342f0643000

Correlation ID: fcac4954-cbe8-4f96-9b13-9f91054d1b7a

Timestamp: 2023-06-23 14:15:26Z

**Comment by tanix12qly** [comment, marked as answer]
直接摸索着解决了，去微软azure设置一个新应用，然后用这个新应用的client id 与 secret，config/api.config.js里面修改为自己刚刚注册的参数就可以。
可以按这个来https://ovi.swo.moe/zh/docs/advanced

**Comment by Duminjun1014** [comment]
解决了，真棒
